internal/interceptortest/tracing.go

```diff
@@ -151,6 +164,8 @@ func AssertSpanPropagation(t *testing.T, tracer TestTracer) {
 				Span(tracer.SpanName(&interceptor.TracerStartSpanOptions{Operation: "RunActivity", Name: "testActivity"}))),
 			Span(tracer.SpanName(&interceptor.TracerStartSpanOptions{Operation: "StartActivity", Name: "testActivityLocal"}),
 				Span(tracer.SpanName(&interceptor.TracerStartSpanOptions{Operation: "RunActivity", Name: "testActivityLocal"})))),
+		// This is the workflow that gets started from the nexus workflow run operation.
+		Span(tracer.SpanName(&interceptor.TracerStartSpanOptions{Operation: "RunWorkflow", Name: "testWaitForCancelWorkflow"})),
```

Comment:
I figure it would have been better to test this with the client but that's not part of our testing setup here. We can consider propagating the parent child relationship in the test env too.

Reply:
Hmmm... I see there's also a tracing integration test, I will add to that.